### PR TITLE
feat: mermaid-support

### PR DESCRIPTION
**Pull request by katiegoines**
#### Description of changes:
New PR with base branched from `next-release/main`. Changes cherry-picked from https://github.com/aws-amplify/docs/pull/5802

Staging link with sample: https://next-release-mermaid-support.d1egzztxsxq9xz.amplifyapp.com/javascript/build-a-backend/api-graphql/

Uses the mermaid rehype plugin to generate inline SVGs of mermaid code blocks into diagrams.

For example the following code
```
```mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
\```
```

Generates the following diagram:
image

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -147,7 +149,7 @@
     "dev": "yarn prebuild && next dev",
     "spellcheck": "cspell \"src/**/*.mdx\" --no-progress",
     "spellcheck-diff": "git diff --name-only --cached | awk \"/src.*\\.mdx/{print}\" | npx cspell --no-must-find-files --file-list stdin",
-    "build": "node tasks/generate-sitemap.mjs && next build && next export -o client/www/next-build",
+    "build": "node tasks/generate-sitemap.mjs && yarn playwright install && next build && next export -o client/www/next-build",
```
Comment (timngyn) [inline package.json:152]: Is there a reason why we don't just install playwright as a dev dependency?

Comment (katiegoines) [reply package.json:152]: It is installed as a dev dependency but it needs to install the browsers on the build.

**Review by jacoblogan — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]

**Review by hbuchel — CHANGES_REQUESTED** [review]
Can we pause on this until after the release? I want us to consider a couple of things:
- the impact of adding Playwright as part of our build step
- how our authors going to ensure accessibility of the graphs they create

**Comment by katiegoines** [comment]
Closing as use of Mermaid needs to be re-addressed.

**Pull request closed without merging**